### PR TITLE
config: fix warnup

### LINKED ISSUE #419: bug: sv_warmup has no effect
I'm not sure if doing something wrong

sv_gametype dm
sv_scorelimit 1
sv_warmup 500

*win*
*no warmup*

### PR DESCRIPTION
**Pull request by ByFox213**
Closes #419.

I honestly have no idea how hosters will use this, since warmup is essentially a end window, and that's very difficult to say. In this version, warmup comes after the end window.

There is a new diff for ddnet!!

### CODE REVIEW

**Comment by ChillerDragon** [comment]
pr_warnup xd

It's war**m**up aaaa :D

File: src/engine/shared/config_variables.h [hunk 1]
```diff
@@ -265,7 +265,8 @@ MACRO_CONFIG_INT(ClShowStartMenuImages, cl_show_start_menu_images, 1, 0, 1, CFGF
 MACRO_CONFIG_INT(ClSkipStartMenu, cl_skip_start_menu, 0, 0, 1, CFGFLAG_CLIENT | CFGFLAG_SAVE, "Skip the start menu")
 
 // server
-MACRO_CONFIG_INT(SvWarmup, sv_warmup, 0, 0, 0, CFGFLAG_SERVER, "Number of seconds to do warmup before round starts")
+// ddnet-insta edited max value
+MACRO_CONFIG_INT(SvWarmup, sv_warmup, 0, 0, 600, CFGFLAG_SERVER, "Number of seconds to do warmup before round starts")
```
Comment (ChillerDragon) [inline src/engine/shared/config_variables.h:269]: I think this diff can be avoided. I am quite confident ddnet would be okay with removing this config on their side so we can add it in the fork and fully control the values.

**Comment by ChillerDragon** [comment]
Warmup and round code drives me crazy there are so many edge cases. Did you test this? I quickly tried to test this in game and did not get it to work but i possibly have a config error somewhere.

I tried `./DDNet-Server "sv_gametype dm;sv_warmup 10"` when I joined nothing happened. Then I tried the rcon command `reload` still no effect.

I tried a vanilla 0.6.5 server as reference which I started using `./teeworlds_srv_d "sv_warmup 10"` and on join as well as on `reload` I got a big warmup timer.

screenshot_2026-07-25_12-04-43

File: src/engine/shared/config_variables.h (hunk 1 above)
Comment (ChillerDragon) [reply src/engine/shared/config_variables.h:269]: For reference the ddnet code matches the 0.6 teeworlds branch

https://github.com/teeworlds/teeworlds/blob/93f5bf632a3859e97d527fc93a26b6dced767fbc/src/game/variables.h#L59

And the current 0.7 master looks like this

https://github.com/teeworlds/teeworlds/blob/5d682733e482950f686663c129adc4b751c8d790/src/game/variables.h#L113

**Comment by ByFox213** [comment]
Nice try from me. But I'm too lazy to do it right now.

Thank you for your efforts!
